api/BucketListAPI.py

```diff
@@ -0,0 +1,329 @@
+
+
+@app.route('/', methods=['GET'])
+def index():
+    """Index route"""
+    return render_template('index.html')
+
+
+@app.route('/auth/register', methods=['POST'])
+def register():
+    """Method to handle user registration"""
+    request.get_json(force=True)
+    try:
+        name = request.json['name']
+        email = request.json['email']
+        password = request.json['password']
+        user = Authenticate()
+        response = user.register(email, password, name)
+        response = auth_success(response)
+        return response
+
+    except KeyError:
+        response = jsonify({'Error': 'Invalid Keys detected'})
+        response.status_code = 500
+        return response
+
+
+@app.route('/auth/login', methods=['POST'])
+def login():
+    """Method to handle user login"""
+    request.get_json(force=True)
+    try:
+        email = request.json['email']
+        password = request.json['password']
+        user = Authenticate()
+        response = user.login(email, password)
+        response = auth_success(response)
+        return response
+
+    except KeyError:
+        response = jsonify({'Error': 'Invalid Keys detected'})
+        response.status_code = 500
+        return response
+
+
+def auth_success(response):
+    if response.status_code == 201:
+        data = json.loads(response.data.decode())
+        data['id'] = encode_auth_token(data['id']).decode()
+        response = jsonify(data)
+        response.status_code = 201
+    return response
+
+
+@app.route('/auth/reset-password', methods=['POST'])
+def reset_password():
+    """Method to handle reset password"""
+    request.get_json(force=True)
+    try:
+        email = request.json['email']
+        old_password = request.json['old_password']
+        new_password = request.json['new_password']
+        user = Authenticate()
+        response = user.reset_password(email, old_password, new_password)
+        return response
+
+    except KeyError:
+        response = jsonify({'Error': 'Invalid Keys detected'})
+        response.status_code = 500
+        return response
+
+
+@app.route('/buckets', methods=['POST'])
+def add_bucket():
+    """Method to handle creating a bucket"""
+    request.get_json(force=True)
+    try:
+        token = get_token()
+        user_id = decode_auth_token(token)
+        if isinstance(user_id, int):
+            bucket_name = request.json['bucket']
+            desc = request.json['desc']
+            bucket = Bucket()
+            response = bucket.create_bucket(bucket_name, desc, user_id)
+            return response
+        return invalid_token()
+
+    except KeyError:
+        response = jsonify({'Error': 'Invalid Keys detected'})
+        response.status_code = 500
+        return response
+
+
+@app.route('/buckets', methods=['GET'])
+def get_buckets():
+    """Method to handle getting all buckets"""
+    try:
+        token = get_token()
+        data = decode_auth_token(token)
+        if isinstance(data, int):
```

Comment:
consider breaking off token validation into a function to avoid repeating same lines of code multiple times